### PR TITLE
lib/http: document client and common

### PR DESCRIPTION
**Pull request by bdilly**
I was working on a few http client samples on C / snippets,
but since I'm investing time on other matters atm,
just sending headers for now.

Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>

### CODE REVIEW

File: src/lib/comms/include/sol-http-server.h [hunk 1]
```diff
@@ -51,14 +52,39 @@ extern "C" {
...
+ * It's created with sol_http_server_new() and should be later
+ * deleted with sol_http_server_del()
+ */
 struct sol_http_server;
+
+/**
+ * @struct sol_http_request
+ * @brief Opaque handler for a request made to a HTTP server.
+ *
+ * Request properties can be gotten with:
```
Comment (mbelluzzo) [inline src/lib/comms/include/sol-http-server.h]: I think s/gotten/queried/ it's better, but it's a matter of taste.

Comment (bdilly) [inline src/lib/comms/include/sol-http-server.h]: ack

**Comment by cabelitos** [comment]
This branch was merged, wasnt it?
Should we close it?

**Comment by bdilly** [comment]
yep, it was merged.
thanks

**Pull request closed without merging**